tests/cli.rs

```diff
@@ -2846,7 +2841,7 @@ fn method_is_changed_when_following_302_redirect() {
             }
             "/second_page" => {
                 assert_eq!(req.method(), "GET");
-                assert!(req.headers().get("Content-Length").is_none());
+                assert_eq!(req.headers().get("Content-Length").unwrap(), "0");
```

Comment:
Currently Hyper for empty bodies with Content-Length header unset automatically sets the value 0

maybe we should change this test case

Reply:
Sounds good to me 👍

Reply:
Is this server-side or client-side? If it's just some weirdness in the test server I'm not worried but if xh is sending this header then I'd like to look into it

Reply:
it is server-side action, hyper server auto set conent-length to 0.
I started a test server to test this feature It will automatically return conent-length: 0


source code:
https://github.com/hyperium/hyper/blob/0013bdda5cd34ed6fca089eceb0133395b7be041/src/proto/h1/role.rs#L881


demo server,you can verify it by run `curl -v http://localhost:3000`

```rust
// tokio = { version = "1.36.0", features = ["full"] }
// hyper = { version = "1.2.0", features = ["server", "http1"] }
// hyper-util = { version = "0.1.3", features = ["full"] }
// http-body-util = { version = "0.1.1",features = [] }

use std::net::SocketAddr;
use http_body_util::Full;
use hyper::body::Bytes;
use hyper::service::service_fn;
use hyper_util::rt::TokioIo;
use tokio::net::TcpListener;

type Builder = hyper_util::server::conn::auto::Builder<hyper_util::rt::TokioExecutor>;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
    let listener = TcpListener::bind(addr).await?;
    loop {
        let (stream, _) = listener.accept().await?;
        let io = TokioIo::new(stream);
        let builder = Builder::new(hyper_util::rt::TokioExecutor::new());
        tokio::task::spawn(async move {
            if let Err(err) = builder
                .clone()
                .serve_connection(
                    io,
                    service_fn( |req| async {
                        hyper::Response::builder()
                            .header("date", "N/A")
                            .header("content-encoding", "gzip")
                            .body::<Full<Bytes>>("".into())
                    }),
                )
                .await
            {
                println!("Error serving connection: {:?}", err);
            }
        });
    }
}

```